onnxruntime/core/common/threadpool.cc

```diff
@@ -304,9 +329,9 @@ int ThreadPool::CurrentThreadId() const {
   return underlying_threadpool_->CurrentThreadId();
 }
 
-Eigen::ThreadPoolInterface* ThreadPool::AsEigenThreadPool() const {
-  ORT_ENFORCE(underlying_threadpool_ != nullptr);
-  return underlying_threadpool_;
+bool ThreadPool::CurrentThreadIsInPool() const {
+  int id = CurrentThreadId();
```

Comment:
What if it is in a pool but is in another pool?

Reply:
This will return true iff the thread is in the current pool.  

Thinking about how to make the API clearer, I realized that the only use of the test is in the combination "CurrentThreadIsInPool() && NumThreads()==1" as a gate for when to use the fast-path sequential code when there are no threads available to help -- so it may be better to pull that overall combined test out as a separate method as opposed to this current test.